### DISCUSSION TITLE
`podman scp` on NixOS

### DISCUSSION THREAD

**Discussion by asbachb** [containers/podman#15397] [General]
Hello,

I'm trying to find out the cause of https://github.com/NixOS/nixpkgs/issues/187589

It seems `pdoman scp` tries to call `podman` command via ssh. I could not find the part in source so that I could verify if this needs to be fixed on podman or on NixOS side.

Based on debug output it's somewhere after here: https://github.com/containers/podman/blob/3dace12777e27a7732acb25ae78f86ced8ae04da/cmd/podman/root.go#L282

Thanks

**Comment by vrothberg** [comment]
Thanks for reaching out!

@cdoern, can you take a look?

**Comment by rhatdan** [comment]
podman scp 
when talking to a remote destination, will attempt an ssh connection and then communicate with a podman service running on the remote machine. Basically it is doing a 
podman save foo.tar
scp foo.tar remote:/tmp/foo.tar
ssh remote podman-remote load foo.tar

**Comment by asbachb** [comment]
@rhatdan When you check the link I posted the problem is that on NixOS this `ssh remote podman-remote load foo.tar` seems to fail. It seems that `podman` local assumes the `podman` binary is present at the remote location in the same place.

That's the reason I look for the code which is doing this remote call to understand if it's a problem of NixOS patching or a podman bug.

**Reply by rhatdan** [reply]
I would say we should just rely on PATH for finding podman on remote machine, if that is what is happening.
@cdoern PTAL.

**Comment by cdoern** [comment]
@asbachb what version of podman are you using? I *believe* I fixed the path issue in the main branch. I will check

**Reply by asbachb** [reply]
@cdoern 4.2 on both sides.

**Reply by asbachb** [reply]
@cdoern Have you had a time to look into?
